datadog/resource_datadog_rum_application.go

```diff
@@ -0,0 +1,138 @@
+package datadog
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceDatadogRUMApplication() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog RUM application resource. This can be used to create and manage Datadog RUM applications.",
+		CreateContext: resourceDatadogRUMApplicationCreate,
+		ReadContext:   resourceDatadogRUMApplicationRead,
+		UpdateContext: resourceDatadogRUMApplicationUpdate,
+		DeleteContext: resourceDatadogRUMApplicationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the RUM application",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "browser",
+				Description: "The RUM application type. Supported values are `browser`, `ios`, `android`, `react-native`, `flutter`",
+			},
+			"client_token": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The RUM application client token",
+			},
```

Comment:
Should we add the computed field `application_id` in the schema as well? Looks like it needs to be used with `client_token`. Or is this same as ID?

Reply:
i think `"The RUM application client token"` is sort of confusing: `RUM Application ID` and `Client Token` are 2 different things and both are required

Only `"The client token"` as description should be better.